janusgraph-core/src/main/java/org/janusgraph/graphdb/configuration/GraphDatabaseConfiguration.java

```diff
@@ -526,10 +529,55 @@ public boolean apply(@Nullable String s) {
             "Whether JanusGraph should attempt to parallelize storage operations",
             ConfigOption.Type.MASKABLE, true);
 
-    public static final ConfigOption<Integer> PARALLEL_BACKEND_OPS_THREAD_POOL_SIZE = new ConfigOption<>(STORAGE_NS,"parallel-backend-ops-thread-pool-size",
-        "Thread pool size which is used for parallel requests when "+PARALLEL_BACKEND_OPS.toStringWithoutRoot()+" is enabled. " +
-            "If not set the pool size will be equal to number of processors multiplied by "+Backend.THREAD_POOL_SIZE_SCALE_FACTOR+".",
-        ConfigOption.Type.MASKABLE, Integer.class);
+    public static final ConfigNamespace PARALLEL_BACKEND_EXECUTOR_SERVICE = new ConfigNamespace(
+        STORAGE_NS,
+        "parallel-backend-executor-service",
+        "Configuration options for executor service which is used for parallel requests when `"+PARALLEL_BACKEND_OPS.toStringWithoutRoot()+"` is enabled.");
+
+    public static final ConfigOption<Integer> PARALLEL_BACKEND_EXECUTOR_SERVICE_CORE_POOL_SIZE = new ConfigOption<>(
+        PARALLEL_BACKEND_EXECUTOR_SERVICE,
+        "core-pool-size",
+        "Core pool size for executor service. May be ignored if custom executor service is used " +
+            "(depending on the implementation of the executor service)."+
+            "If not set the core pool size will be equal to number of processors multiplied by "
+            +ExecutorServiceBuilder.THREAD_POOL_SIZE_SCALE_FACTOR+".",
+        ConfigOption.Type.LOCAL,
+        Integer.class);
+
+    public static final ConfigOption<Integer> PARALLEL_BACKEND_EXECUTOR_SERVICE_MAX_POOL_SIZE = new ConfigOption<>(
+        PARALLEL_BACKEND_EXECUTOR_SERVICE,
+        "max-pool-size",
+        "Maximum pool size for executor service. Ignored for `fixed` executor service and may be ignored " +
+            "if custom executor service is used (depending on the implementation of the executor service)."+
+            "If not set the core pool size will be equal to number of processors multiplied by "
+            +ExecutorServiceBuilder.THREAD_POOL_SIZE_SCALE_FACTOR+".",
```

Comment:
I think it should be "If not set, the _max_ pool size will be Integer.MAX_VALUE."

Reply:
I think we probably don't even need to have this statement because the default value is present in the documentation. So, probably it's just unnecessary to duplicate that information. I removed this statement